bulk-load/src/main/java/no/sikt/nva/data/report/api/etl/transformer/GenerateKeyBatchesHandler.java

```diff
@@ -0,0 +1,150 @@
+package no.sikt.nva.data.report.api.etl.transformer;
+
+import static java.util.Objects.nonNull;
+import static java.util.UUID.randomUUID;
+import com.amazonaws.services.lambda.runtime.Context;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.List;
+import java.util.stream.Collectors;
+import no.unit.nva.events.handlers.EventHandler;
+import no.unit.nva.events.models.AwsEventBridgeEvent;
+import nva.commons.core.Environment;
+import nva.commons.core.JacocoGenerated;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.eventbridge.EventBridgeClient;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequest;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequestEntry;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsResponse;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Request;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Response;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.S3Object;
+
+public class GenerateKeyBatchesHandler extends EventHandler<KeyBatchRequestEvent, Void> {
+
+    public static final String DEFAULT_BATCH_SIZE = "1000";
+    public static final String DELIMITER = "/";
+    public static final String DEFAULT_START_MARKER = null;
+    public static final String START_MARKER_MESSAGE = "Start marker: {}";
+    private static final Logger logger = LoggerFactory.getLogger(GenerateKeyBatchesHandler.class);
+    private static final Environment ENVIRONMENT = new Environment();
+    public static final String INPUT_BUCKET = ENVIRONMENT.readEnv("EXPANDED_RESOURCES_BUCKET");
+    public static final String OUTPUT_BUCKET = ENVIRONMENT.readEnv("KEY_BATCHES_BUCKET");
+    public static final String EVENT_BUS = ENVIRONMENT.readEnv("EVENT_BUS");
+    public static final String TOPIC = ENVIRONMENT.readEnv("TOPIC");
+    public static final String MANDATORY_UNUSED_SUBTOPIC = "DETAIL.WITH.TOPIC";
+    public static final int MAX_KEYS = Integer.parseInt(
+        ENVIRONMENT.readEnvOpt("BATCH_SIZE").orElse(DEFAULT_BATCH_SIZE));
+    private static final String DEFAULT_LOCATION = "resources";
```

Comment:
Do you need functionality to read from other s3 locations than resources? nvi-candidates?

Reply:
yes.